hw/ip/csrng/doc/_index.md

```diff
@@ -0,0 +1,89 @@
+1. An AES primitive
+2. The CTR_DRBG state-machine (CTR_DRBG_FSM) which drives the AES primitive, performing the various encryption sequences proscribed in SP 800-90A.
+These include:
+
+    1. **The Derivation Function:**  
+       Part of the instantiation and reseed routines, this routine assembles the previous seed material (on reseed only), application inputs, and entropy.
+    2. **The Instantiation Routine:**
+       Combines application inputs, external entropy and nonce (more entropy) via the derivation function.
+    3. **The Reseed Routine:**
+       Combines the previous seed material with external entropy to generate a new seed.
+    4. **The Generate Routine:**
+       Generates up to CSRNG_MAX_GENERATE random bits.
+       If called with prediction_resistance_flag, forces a reseed.
+     5. **The Update Routine:**
+        Updates the internal state of the DRBG instance after each generate call.
+3. State vectors for each DRNG instance.
+4. Interface logic and access control for each instance.
+
+## Note on the term "Entropy"
+
+Every DRNG requires some initial seed material, and the requirements for the generation of that seed material varies greatly between standards, and potentially between CC security targets.
+In all standards considered, DRNG's require some "entropy" from an external source to create the initial seed.
+However, the rules for obtaining said entropy differ.
+Furthermore the required delivery mechanisms differ.
+For this reason we must make a clear distinction between "Physical" (or "Live" or "True") entropy and "Factory Entropy".
+This distinction is most important when considering the creation of IP which is both compatible with  IPS 140-3 recommendations, and FIPS 140-2 use cases (which have already been certified for a variety of applications).
+
+- Physical entropy is the only type of "entropy" described in SP 800-90.
+The means of generation is described in SP 800-90B.
+One statistical test requirement is that physical entropy must be unique between reboot cycles, ruling out sources such as one-time programmable (OTP) memories.
+In SP 800-90A, the delivery mechanism must come through a dedicated interface and "not be provided by the consuming application".
+
+- Factory entropy is a type of entropy described in the [FIPS 140-2 implementation guidance (IG)](https://csrc.nist.gov/csrc/media/projects/cryptographic-module-validation-program/documents/fips140-2/fips1402ig.pdf).
+It can be stored in a persistent memory, programmed at the factory.
+In some use cases, the consuming application needs to explicitly load this entropy itself and process it to establish the expected seed.
+(However this is permissible in FIPS 140-2.)
+
+This document aims to make the distinction between physical entropy and factory entropy wherever possible.
+However, if used unqualified, the term "entropy" should be understood to refer to physical entropy strings which are obtained in accordance with SP 800-90C.
```

Comment:
I think this is the spot to link to the specific 800-90C PDF you are referring to.

(As I look at this more, I could be convinced that we could have a final section with external references in this document?  Just to avoid any confusion about which versions we are using, and then the document-internal refs could just point to the references section?  WDYT?)

Reply:
Made a reference to SP 800-90C Second Draft at the same point my first reference to the term NRBG above.   As far as the links to the references section that sounds nice.  However is there a reference as how to do that in our other Markdown documents?  I'll take a look around.